### PR TITLE
Bl 724 BE: I want to register as an instructor

### PR DESCRIPTION
**Pull request by tkordysh**
## Description
Updated oktaClient file and .env variables to allow for user creation. Added post functionality to instructor router to allow for instructor registration with proper role id. 

[## Loom Video]
https://www.loom.com/share/8837231f0c7e4cc98e3af1fcf90c122f

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by vasilii-garanin — DISMISSED** [review]
Hi guys, good job with updates in back end to create new instructor. I think codeclimate check is not happy because newInstructor looks the same as newUser. I would think that's ok but might check in with Release Manager to double check. If I could give a suggestion on an easier way to refresh your database, there is a button that looks like triangular "play" button. If you click that button it will refresh your query.

**Review by Irving-Delgad0 — DISMISSED** [review]
great job on the code! this looks great and goes well with the front end for instructor registration

**Review by AnthonyLayne — DISMISSED** [review]
Your code looks good. The only thing I would recommend is getting rid of the console.logs in the userRouter.js file. Great Job!

**Review by gclancy121 — APPROVED** [review]
This looks good to me, no obvious bugs like spelling errors

**Review by gclancy121 — DISMISSED** [review]
Personally I would remove the console.log()s

**Review by Roadlyfe — DISMISSED** [review]
Looks great! Clean code.

**Review by dshipuk — DISMISSED** [review]
Great job! Code looks good

File: api/instructor/instructorRouter.js
Comment (paulstgermain) [inline api/instructor/instructorRouter.js:30]: We'll want to add a `pending` column here and in our migration and seeds for profiles. For instructors, it will default to `true`, to be changed to `false` when their application has been accepted through the admin's dashboard.

For all other users, so in `userRouter.js`, we'll default to `false`, as their registration will simply be accepted by default.

**Review by paulstgermain — CHANGES_REQUESTED** [review]
This is some really solid implementation here, and I'm excited to get this much closer to creating Okta users from within our app's registration process!

That said, there are a couple of things I'd love to address before we can merge this PR up:

File: api/instructor/instructorRouter.js
Comment (paulstgermain) [inline api/instructor/instructorRouter.js]: We also don't want to be attempting to create a user with a password, I don't think! Okta should allow us to create a user without a given password, emailing them to have them complete their registration and pick their password then.

**Review by reneetro — DISMISSED** [review]
Looks nice and clean! Were the console logs intentionally left in the userRouter? The comments are also really helpful, thank you for adding those!

**Review by Cassius-Cassity — DISMISSED** [review]
Code looks clean to me all I would do is get rid of all the console.logs.

File: api/instructor/instructorRouter.js
Comment (tkordysh) [reply api/instructor/instructorRouter.js:30]: Added these changes!

File: api/instructor/instructorRouter.js
Comment (tkordysh) [reply api/instructor/instructorRouter.js]: Added these changes! Also, cleaned up console.logs

**Review by Roadlyfe — APPROVED** [review]
This looks great! Clean code with no obvious bugs.

**Review by Cassius-Cassity — APPROVED** [review]
Code looks good, not seeing any obvious mistakes.

**Review by ryan-hamblin — DISMISSED** [review]
Great work here. Your code follows industry practices for setting up routes and publishing data to your persisted layer (in this case PG). I really liked how in you structured the body and newUser object inside of `InstructorRouter.js`. One way to optimize that for the future could be to destructure body using the `const { thing } = req.body` syntax, however I like how you got it done the JS 2015 way!

One thing I will request you remove is the Env Vars values found in the PR Description. This is a massive security flaw. 

1.) I need to recycle that key that was leaked out on the internet.
2.) please delete those instances from the description
3.) ensure that you document the new env vars in our product documents for future learners.

**Review by paulstgermain — APPROVED** [review]

**Review by AlexiusThomas — COMMENTED** [review]
Looks good to me, no obvious bugs or errors

**Pull request merged**
